### PR TITLE
tables: escape double quotes for C# compatibility

### PR DESCRIPTION
**Pull request by AliceInHunterland**
Ref https://github.com/nspcc-dev/neo-go/issues/3284.

### CODE REVIEW

File: encode.go
Comment (AnnaShaleva) [inline encode.go:917]: Need to adjust comment to `default` case.

**Review by AnnaShaleva — COMMENTED** [review]
Good, the fix itself looks legit, need to fix the rest of tests. I'll check the mainnet with it.

**Comment by AnnaShaleva** [comment]
We also need to check that deserialization worksproperly.

**Comment by AnnaShaleva** [comment]
BTW, wrong issue reference.

File: encode.go
Comment (AnnaShaleva) [inline encode.go:930]: It's a single quote.

File: encode_test.go
Comment (AnnaShaleva) [inline encode_test.go:234]: Useless change.

File: encode_test.go
Comment (AnnaShaleva) [inline encode_test.go:614]: Not related to this repo, we'd better have this test in NeoGo.

**Review by AnnaShaleva — COMMENTED** [review]
Otherwise LGTM.

**Review by AnnaShaleva — APPROVED** [review]
Good, waiting for mainnet/testnet states confirmation before the merge.

**Review by AnnaShaleva — CHANGES_REQUESTED** [review]
Adjust commit message wrt PR title and change issue in the commit body.

**Comment by AliceInHunterland** [comment]
> Good, waiting for mainnet/testnet states confirmation before the merge.

```
ekaterina@ekaterina-ThinkPad-T14-Gen-1:~/Desktop/neo-go$ go run scripts/compare-states/compare-states.go http://localhost:10332 http://seed1.neo.org:10332 
at 0: 58a5157b7e99eeabf631291f1747ec8eb12ab89461cda888492b17a301de81e8 vs 58a5157b7e99eeabf631291f1747ec8eb12ab89461cda888492b17a301de81e8
at 4716143: a64d3eaec83ab6c45f34136bcb4473d0f12821c283440f297e2c69fdc1d78886 vs a64d3eaec83ab6c45f34136bcb4473d0f12821c283440f297e2c69fdc1d78886
```

```
ekt@ekt:~/neo-go$ go run ./scripts/compare-states/compare-states.go http://localhost:20332 http://seed1t5.neo.org:20332
at 0: 62fd8ff9b0543aea352257db5b00bbb01d1bc0d2cc665e1f24cf5de0d16ebc7b vs 62fd8ff9b0543aea352257db5b00bbb01d1bc0d2cc665e1f24cf5de0d16ebc7b
at 3320969: 7cbac469505835bd8d9c6a5bc84172d4234b0d04235b936ee3f391bff6120e29 vs 7cbac469505835bd8d9c6a5bc84172d4234b0d04235b936ee3f391bff6120e29
```

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**
